### PR TITLE
build(binding): MacOS ARM64 native binaries

### PR DESCRIPTION
**Pull request by rexfleischer**
# Description

Change to add the native binaries for Mac OS ARM64. This is based on #112 and #190. I am not able to test all the different binaries, but I was able to test the non-freetype macos-arm64.

## Type of change

- [X] Minor changes or tweaks (quality of life stuff)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

### CODE REVIEW

**Comment by Displee** [comment]
Yes please! I need this so bad!

**Comment by zly2006** [comment]
kindly reminds that you could use `* [X]` instead of `* [ X ]` (remove whitespaces) so it looks like ☑️ in your description

File: imgui-binding/src/main/java/imgui/ImGui.java [hunk 1]
```diff
@@ -29,7 +29,8 @@
 public class ImGui {
     private static final String LIB_PATH_PROP = "imgui.library.path";
     private static final String LIB_NAME_PROP = "imgui.library.name";
-    private static final String LIB_NAME_DEFAULT = System.getProperty("os.arch").contains("64") ? "imgui-java64" : "imgui-java";
+    private static final boolean isARM = System.getProperty("os.arch").equals("arm") || System.getProperty("os.arch").startsWith("aarch64");
+    private static final String LIB_NAME_DEFAULT = "image-java" + (isARM ? "arm" : "") + "64";
```
Comment (zly2006) [inline imgui-binding/src/main/java/imgui/ImGui.java:33]:
```suggestion
    private static final String LIB_NAME_DEFAULT = "imgui-java" + (isARM ? "arm" : "") + "64";
```

Hey this is "imgui" not "image" !

**Review by zly2006 — CHANGES_REQUESTED** [review]

**Review by zly2006 — CHANGES_REQUESTED** [review]
Also have you checked the imgui-app-all jar? There is some issues in the build script:
- arm64 nativa lib is not found in /bin folder
- the lib is also not found in `:imgui-app:shadowJar` task (reason might be above)

File: imgui-binding/src/main/java/imgui/ImGui.java (hunk 1 above)
Comment (rexfleischer) [reply imgui-binding/src/main/java/imgui/ImGui.java:33]: oh crap...

**Comment by rexfleischer** [comment]
I haven't checked the all one because I didn't know there was one! I will do that as well. Thanks for the reviews.

**Comment by rexfleischer** [comment]
Update will be in soon.

Ok, this was a couple months ago so I'm not sure if I'm remembering correctly. Here are some notes (and maybe a follow up question):
* I originally assumed that I didn't need to include the built natives in bin. I forgot why I thought that... However, I'm copying them into bin now. Is there a standard way of doing this that I'm missing? I just got the `./gradlew imgui-binding:generateLibs -Denvs=macosarm64 -Dfreetype=true/false` working and copied `/tmp/imgui/libsNative/macosxarm64/libimgui-javaarm64.dylib` to bin.
* I couldn't get the freetype build working until I executed the `FreeType - Install (MacOS-arm64)` from `.github/workflows/ci.yaml` step. This sound reasonable?
* After executing `./gradlew :imgui-app:shadowJar`, I executed `jar tf imgui-app/build/libs/imgui-app-686c27-all.jar` and now see the libimgui-javaarm64.dylib library. 

I do have one follow up question. I see in the `image-app` build script for the jar, it copies in the non-freetype linux and mac natives, but only the freetype windows native. What is the reason for this?

**Comment by zly2006** [comment]
> However, I'm copying them into bin now. Is there a standard way of doing this that I'm missing?

There is a github ci script for that: `.github/workflows/ci.yaml`

> * I couldn't get the freetype build working until I executed the `FreeType - Install (MacOS-arm64)` from `.github/workflows/ci.yaml` step. This sound reasonable?

There's an wrong if condition, I fixed this issue in #190, merge the upstream to fix

> * After executing `./gradlew :imgui-app:shadowJar`, I executed `jar tf imgui-app/build/libs/imgui-app-686c27-all.jar` and now see the libimgui-javaarm64.dylib library.

I am not quite sure... I have not looked carefully into your code.

> I do have one follow up question. I see in the `image-app` build script for the jar, it copies in the non-freetype linux and mac natives, but only the freetype windows native. What is the reason for this?

see readme: https://github.com/SpaiR/imgui-java?tab=readme-ov-file#freetype
freetype was better in rendering, and there is no dependency issues on windows (unlike linux and macos you should install freetype first) 

After all, your PR still cannot be compiled and seems far away from being merged, I am neither the maintainer of this project nor an expert in imgui, I am not asking you to close the pr but it really has some problems and I think #190 works better.

**Comment by rexfleischer** [comment]
I opened this because both #112 and #190 seemed to be abandoned and had either changes requested by the maintainer or had other issues. If you can get 190 in, I don't care either way. However, 190 still has changes requested by the maintainer in 112. 

> your PR still cannot be compiled

how do you figure this? It compiles: https://github.com/rexfleischer/imgui-java/actions/runs/9118399802

> seems far away from being merged

If you "neither the maintainer of this project nor an expert in imgui", how do you figure this? I'm sorry if this PR caused some turmoil, but both the other PRs seemed far away from being merged.

**Comment by rexfleischer** [comment]
Either way, @SpaiR if there is anything that can help move this PR or #190 forward, please let us know.

**Comment by SpaiR** [comment]
First of all, I apologize for the inconvenience to everyone who was waiting for the macOS ARM support and specifically for this PR to be merged. 🙏 

There were several PRs, but for various reasons, I wasn't able to review them until now. However, the time has finally come!

I will merge this PR since it includes some minor CI cleanups. I also want to thank everyone who has contributed (#112 and #190). I will mention all authors in the upcoming changelog, which I believe will be released very soon (within 1-2 weeks or even sooner).

**Comment by SpaiR** [comment]
Ok, I revamped the whole concept. In PR #239, a universal binary for the ARM64 architecture was introduced. So, one dylib for all - which is great, I believe. Additionally, while working on the build scripts, I decided to resolve the longstanding issue with FreeType. As a result, FreeType is now statically compiled and enabled by default, which basically improves font quality for free. More details can be found in the PR description.

**Pull request merged**